src/react/components/UserProvider.tsx

```diff
@@ -46,22 +55,56 @@ export const UserProvider = (props: Props) => {
   const [accessToken, setAccessToken] = useState<string | null>(null);
   const [isLoading, setIsLoading] = useState<boolean>(!initialUser);
   const [error, setError] = useState<Error>();
+  const [refreshToken, setRefreshToken] = useState<RefreshToken | null>(null);
+  const refreshTimer = useRef<NodeJS.Timeout | undefined>();
   const { pathname } = useRouter();
 
   const checkSession = useCallback(async (): Promise<void> => {
     try {
-      const { user, accessToken } = await fetcher(profileUrl);
+      const { user, accessToken, refreshToken, expiresAt } = await fetcher(
+        profileUrl
+      );
       if (accessToken) {
         supabaseClient.auth.setAuth(accessToken);
         setAccessToken(accessToken);
       }
+      if (refreshToken && expiresAt) {
+        setRefreshToken({ token: refreshToken, expiresAt });
+      }
       setUser(user);
       if (!user) setIsLoading(false);
```

Comment:
Seems like there's some double-handling of `isLoading` in this component – it's set in `checkSession`, and in the two `useEffect` hooks that call `checkSession`. Seems worth cleaning up, but I didn't want to muddy this PR.

Reply:
While I'm thinking about it, it seems strange to set `isLoading` to `true` on subsequent reloads of the user (after the first load). If you think about how `isLoading` would be used practically, it'd be to show loading states etc on an initial page load. As a consumer of the hook, I wouldn't expect `isLoading` to then flick back to false then true again every time the user is refreshed – it'd cause loading states to briefly pop back in, which would be strange